### PR TITLE
DeliveryreportController and statusService

### PR DESCRIPTION
**Pull request by tba76**
## Related Issue(s)
- Altinn/altinn-notifications#377

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Notifications.Sms.Core/Status/IStatusService.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/// <summary>
+/// Describes the required public method of the status service.
+/// </summary>
+public interface IStatusService
+{
+    /// <summary>
+    /// Update the status of an sms
+    /// </summary>
+    /// <param name="message">DeliveryReport from Link Mobility</param>
+    /// <returns></returns>
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Sms.Core/Status/IStatusService.cs:14]: empty returns

File: src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs [hunk 2]
```diff
@@ -1,28 +1,72 @@
...
+
+    /// <summary>
+    /// No delivery report received from operator. Unknown delivery status.
+    /// </summary>
+    NULL,
+
+    /// <summary>
+    /// The receiver number is barred/blocked/not in use. Do not retry message, and remove number from any subscriber list.
+    /// </summary>
+    BARRED
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs:71]: Could this rather be Failed_BarredReceiver ?

File: src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs [hunk 3]
```diff
@@ -1,28 +1,72 @@
...
+
+    /// <summary>
+    /// Message was rejected.
+    /// </summary>
+    REJECTED,
+
+    /// <summary>
+    /// The SMS failed to be delivered because no operator accepted the message or due to internal Gateway error.
+    /// </summary>
+    FAILED,
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs:61]: we already have a failed status

File: src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs (hunk 2 above)
Comment (tba76) [reply src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs]: Yes, changed all fail-statuses.

File: src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs (hunk 3 above)
Comment (tba76) [reply src/Altinn.Notifications.Sms.Core/Status/SmsSendResult.cs]: Removed

File: src/Altinn.Notifications.Sms/Program.cs [hunk 4]
```diff
@@ -87,6 +107,14 @@ async Task ConnectToKeyVaultAndSetApplicationInsights(ConfigurationManager confi
 
 void ConfigureServices(IServiceCollection services, ConfigurationManager configuration)
 {
+    UserSettings userSettings = configuration!.GetSection(nameof(UserSettings)).Get<UserSettings>()!;
+
+    if (userSettings == null)
+    {
+        throw new ArgumentNullException(nameof(configuration), "Required User settings is missing from application configuration");
```
Comment (acn-sbuad) [inline src/Altinn.Notifications.Sms/Program.cs:114]:
```suggestion
        throw new ArgumentNullException(nameof(configuration), "Required user settings is missing from application configuration");
```

**Review by acn-sbuad — APPROVED** [review]

**Pull request merged**